schemas/tests.yaml

```diff
@@ -0,0 +1,158 @@
+---
+
+#
+# JSON Schema definition for tmt `Tests`
+# https://tmt.readthedocs.io/en/stable/spec/tests.html
+#
+
+definitions:
+
+  beakerlib_library:
+    type: array
+    minItems: 1
+
+    items:
+      oneOf:
+        # library(repo/lib)
+        - type: string
+          pattern: "^library([a-zA-Z0-9]+/[a-zA-Z0-9]+)"
+
+        # url: https://github.com/beakerlib/openssl
+        # name: /certgen
+        - type: object
+
+          properties:
+            name:
+              type: string
+
+            url:
+              type: string
+              pattern: "^https?://"
+
+          required:
+            - name
+            - url
+
+        # path: /tmp/library/openssl
+        # name: /certgen
+        - type: object
+
+          properties:
+            name:
+              type: string
+
+            path:
+              type: string
+              pattern: "^/.+"
+
+          required:
+            - name
+            - path
+
+        # url: https://github.com/redhat-qe-security/certgen
+        # ref: devel
+        # nick: openssl
+        # name: /certgen
+        - type: object
+
+          properties:
+            name:
+              type: string
+            nick:
+              type: string
+            ref:
+              type: string
+            url:
+              type: string
+              pattern: "^https?://"
+
+          required:
+            - name
+            - nick
+            - ref
+            - url
+
+  string_or_array:
+    oneOf:
+      - type: string
+
+      - type: array
+        minItems: 1
+
+$id: tests.yaml
+
+$schema: https://json-schema.org/draft-07/schema
+
+type: object
+
+additionalProperties: false
+
+properties:
+
+  # TODO: adjust
+
+  component:
+    $ref: "tests.yaml#/definitions/string_or_array"
+
+  contact:
+    $ref: "tests.yaml#/definitions/string_or_array"
+
+  description:
+    type: string
+
+  duration:
+    type: string
+    pattern: "^[0-9]+[smhd]$"
+
+  enabled:
+    type: string
+    pattern: "^(true|false)$"
+
+  environment:
+    type: object
+
+    minProperties: 1
+
+    additionalProperties:
+      type: string
+
+  framework:
+    type: string
+    pattern: "^(beakerlib|shell)$"
+
+  # TODO: link
+
+  manual:
+    type: string
+    pattern: "^(true|false)$"
+
+  order:
+    type: integer
+    multipleOf: 1
+
+  path:
+    type: string
+    pattern: "^/.+"
+
+  recommend:
+    $ref: "tests.yaml#/definitions/string_or_array"
+
+  require:
+    oneOf:
+      - type: string
+      - $ref: "tests.yaml#/definitions/beakerlib_library"
+
+  summary:
+    type: string
+
+  tag:
+    $ref: "tests.yaml#/definitions/string_or_array"
+
+  test:
+    type: string
+
+  tier:
+    type: string
```

Comment:
From docs:

> Should be a string.

But it's often an integer, both in real-world and the example in docs. I'd allow `integer` here as well as string, updating the docs to reflect reality, and normalization in my patch takes care of converting any integer into a string.

Reply:
added integer oneof